pkg/domain/rates/rate-fetch_test.go

```diff
@@ -0,0 +1,74 @@
+package rates
+
+import (
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"net/http/httptest"
+	"se-school-case/pkg/util/constants"
+	"testing"
+)
+
+// Create mock fetch server
+func mockRateAPIServer(responseBody string, statusCode int) *httptest.Server {
```

Comment:
Can we refactoring this test in table test manner ? It's mostly recommendation for working with table tests.